test/base.py

```diff
@@ -15,23 +15,52 @@
 
 import dataclasses
 import json
-from typing import Any, Callable, Optional
 import warnings
+from typing import Any, Callable, Optional
 
 import numpy as np
 from qiskit.test import QiskitTestCase
+from qiskit_experiments.calibration_management import Calibrations
+from qiskit_experiments.database_service.db_experiment_data import ExperimentStatus
 from qiskit_experiments.framework import (
     ExperimentDecoder,
     ExperimentEncoder,
+    ExperimentData,
     BaseExperiment,
     BaseAnalysis,
 )
-from qiskit_experiments.calibration_management import Calibrations
 
 
 class QiskitExperimentsTestCase(QiskitTestCase):
     """Qiskit Experiments specific extra functionality for test cases."""
 
+    def assertComplete(self, experiment_data: ExperimentData):
+        """Assert that an experiment is succeeded.
+
+        Args:
+            experiment_data: Experiment data to evaluate.
+        """
+        experiment_data.block_for_results()
+
+        self.assertEqual(
+            experiment_data.status(),
+            ExperimentStatus.DONE,
+            msg="All threads are executed but status is not DONE. " + experiment_data.errors(),
+        )
+
+    def assertFail(self, experiment_data: ExperimentData):
```

Comment:
This one is a little broad, since its only checking if an experiment is not done. If you really want to check for a fail you could change the check to be for `ExperimentStatus.ERROR` so it doesn't  pass for t things like `RUNNING` `POST_PROCESSING` etc.

That said, it doesn't seem that beneficial to have unique asserts for these anymore since they are basically a 1 line call to `assertEqual(experiment.status(), target_status)`, but with implicit blocking. The blocking should be mentioned in the test doc string since it could break a other tests if you need to check things without blocking (eg a minor thing is all the other tests that relied on analysis_results blocking now don't test that since the blocking is happening here).